proto/cosmos/accesscontrol/accesscontrol.proto

```diff
@@ -18,6 +18,18 @@ message WasmAccessOperation {
     string selector = 3;
 }
 
+message WasmContractReference {
+    string contract_address = 1;
+    WasmMessageSubtype message_type = 2;
```

Comment:
nit we can probably get rid of this field by having `contract_query_references` and `contract_execute_references` in `WasmContractReferences` to save a few bytes and some if statements later on in code

Reply:
This is different, because the contract_query_references and contract_execute_references specify what branch to take for contract A based on the incoming WasmMessageInfo. This message type specifies the message type to send to the contract B GetWasmDependency and building that WasmMessageInfo

Reply:
I'm thinking of each `WasmContractReferences` corresponds to one message type of contract A, and its fields `contract_query_references` would be a list of `<contract X, some message type of contract X>` tuple, and likewise for execute. Would that work?

Reply:
I'm thinking of each WasmContractReference as a pointer to the contract B or C or whatever and what message to send to that contract. Especially relevant when we add in the changes that allow for JSON translation, since if we have a list of contract X with message type of contract X, we would also need the JSON translation template too, so it make more sense to me to have WasmContractReference be containing all of the information for calling another contract